### PR TITLE
Add new duration field to JSON output #1937

### LINKED ISSUE #1937: add duration of the scan to JSON
## Short Description

currently there is already a start time stamp and end time stamp in the JSON output. It would be nice to also have a "duration" field so I don't have to compute that myself.

## Possible Labels

- new feature

## Select Category

-  Enhancement [X]
-  Add License/Copyright []
-  Scan Feature []
-  Packaging []
-  Documentation []
-  Expand Support []
-  Other []

## **Possible Solution/Implementation Details**

end_timestamp - start_timestamp

### PR DESCRIPTION
**Pull request by MankaranSingh**
Fixes #1937

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by MankaranSingh** [comment]
@pombredanne Sorry for inconvenience, but i thought it was better to open a new clean pull request instead of fixing https://github.com/nexB/scancode-toolkit/pull/1939  
I have made all required changes ! thanks for you guidance !

File: src/scancode/cli.py [hunk 1]
```diff
@@ -949,8 +949,11 @@ def echo_func(*_args, **_kwargs):
         codebase.counters['final:files_count'] = files_count
         codebase.counters['final:dirs_count'] = dirs_count
         codebase.counters['final:size_count'] = size_count
+        
+        processing_end = time()
```
Comment (pombredanne) [inline src/scancode/cli.py]: Why do you need an intermediate variable?
May be you could directly use `time()` in https://github.com/nexB/scancode-toolkit/pull/1942/files#diff-1e37ecfb9a8c3fccd8ecdfb2ce51f591R956 ?

**Comment by MankaranSingh** [comment]
@pombredanne i did the required changes, hope they are ok :)
also sorry for delays as my exams are going on and i get free only around this time of the night.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -11,6 +11,8 @@ Minor bug fixes:
  - Set proper links in the README. Reported and fixed by @sschubert #1371
  - No changes from v3.0.1
 
+Outputs:
```
Comment (pombredanne) [inline CHANGELOG.rst]: Please put that at the top for now, as this was NOT in 3.0.2

File: src/scancode/cli.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#
```
Comment (pombredanne) [inline src/scancode/cli.py]: Why this change?

File: tests/scancode/test_cli.py [hunk 4]
```diff
@@ -722,6 +722,15 @@ def test_scan_errors_out_with_conflicting_verbosity_options():
...
 
+def test_scan_valid_duration_field_in_json_output_headers():
+    test_file = test_env.get_test_loc('license_text/test.txt')
+    result_file = test_env.get_temp_file('results.json')
+    args = ['--json', result_file, test_file]
+    run_scan_click(args)
+    with open(result_file) as result:
+        headers = json.loads(result.read())['headers']
+    assert headers[0]['duration']
+    assert headers[0]['duration'] >= 0
```
Comment (pombredanne) [inline tests/scancode/test_cli.py]: Excellent!

File: src/scancode/cli.py (hunk 3 above)
Comment (MankaranSingh) [reply src/scancode/cli.py]: Must had been a typo.

**Comment by MankaranSingh** [comment]
@pombredanne all the required changes done :)
